component/src/main/java/org/wso2/extension/siddhi/execution/geo/GetCityResolverFunction.java

```diff
@@ -0,0 +1,216 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.execution.geo;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.extension.siddhi.execution.geo.api.GeoLocationResolver;
+import org.wso2.extension.siddhi.execution.geo.internal.LRUCache;
+import org.wso2.extension.siddhi.execution.geo.internal.exception.GeoLocationResolverException;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.ReturnAttribute;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.function.FunctionExecutor;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.Map;
+
+/**
+ * The class provides implementation for getting city based on the given ip.
+ */
+@Extension(
+        name = "findCityFromIP",
+        namespace = "geo",
+        description = "Returns the city which is related to the give ip address.",
+        parameters = {
+                @Parameter(
+                        name = "ip",
+                        description = "The IP address that the user need to get the relevant city",
+                        type = {DataType.STRING})
+        },
+        returnAttributes = @ReturnAttribute(
+                description = "The city which is related to the given IP",
+                type = {DataType.STRING}),
+        examples = @Example(
+                description = "This will return the corresponding city to the given ip address",
+                syntax = "define stream IpStream(ip string);\n" +
+                        "from IpStream\n" +
+                        "select geo:getCity(ip) as city\n" +
+                        "insert into outputStream;")
+)
+public class GetCityResolverFunction extends FunctionExecutor {
+    private static final Log log = LogFactory.getLog(GetCityResolverFunction.class);
+
+    private static final String CACHE_SIZE_KEY = "cacheSize";
+    private static final String IS_CACHE_ENABLED = "isCacheEnabled";
+    private static final int DEFAULT_CACHE_SIZE = 10000;
+    private static final String DEFAULT_GEOLOCATION_RESOLVER_CLASSNAME =
+            "org.wso2.extension.siddhi.execution.geo.internal.impl.DefaultDBBasedGeoLocationResolver";
+
+    private GeoLocationResolver geoLocationResolverImpl;
+    private static LRUCache<String, String> cityLRUCache;
```

Comment:
Having a static cache is correct by move that to a common class so that we don't need to have 2 instances for both the find city and find country functions